CMakeLists.txt

```diff
@@ -163,5 +166,22 @@ if(SOCI_TESTS)
   add_subdirectory(tests)
 endif()
 
+###############################################################################
+# build config file
+###############################################################################
```

Comment:
Now I handle only cache variable starting with `SOCI_HAVE_`. I'm using CACHE_HELPSTRING property to add some help in generated header